addons/binding/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/handler/HomematicBridgeHandler.java

```diff
@@ -0,0 +1,307 @@
+
+            gateway = HomematicGatewayFactory.createGateway(id, config, this);
+            gateway.initialize();
+
+            registerDeviceDiscoveryService();
+            scheduler.submit(new Runnable() {
+
+                @Override
+                public void run() {
+                    discoveryService.waitForScanFinishing();
+                    updateStatus(ThingStatus.ONLINE);
+                    for (Thing hmThing : getThing().getThings()) {
+                        hmThing.getHandler().thingUpdated(hmThing);
+                    }
+                }
+            });
+
+        } catch (IOException ex) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, ex.getMessage());
+            logger.error(ex.getMessage(), ex);
+            dispose();
+            scheduleReinitialize();
+        }
+    }
+
+    /**
+     * Schedules a reinitialization, if the Homematic gateway is not reachable at bridge startup.
+     */
+    private void scheduleReinitialize() {
+        scheduler.schedule(new Runnable() {
+
+            @Override
+            public void run() {
+                initialize();
+            }
+        }, REINITIALIZE_DELAY_SECONDS, TimeUnit.SECONDS);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void dispose() {
+        logger.debug("Disposing bridge '{}'", getThing().getUID().getId());
+        super.dispose();
+        if (discoveryService != null) {
+            discoveryService.stopLoadDevices();
+            unregisterDeviceDiscoveryService();
+        }
+        if (gateway != null) {
+            gateway.dispose();
+        }
+        if (config != null) {
+            portPool.release(config.getCallbackPort());
+        }
+    }
+
+    /**
+     * Registers the DeviceDisoveryService.
+     */
+    private void registerDeviceDiscoveryService() {
+        logger.trace("Registering HomematicDeviceDiscoveryService for bridge '{}'", getThing().getUID().getId());
+        discoveryService = new HomematicDeviceDiscoveryService(this);
+        discoveryServiceRegistration = bundleContext.registerService(DiscoveryService.class.getName(), discoveryService,
+                new Hashtable<String, Object>());
+        discoveryService.activate();
+    }
+
+    /**
+     * Unregisters the DeviceDisoveryService.
+     */
+    private void unregisterDeviceDiscoveryService() {
+        if (discoveryServiceRegistration != null) {
+            HomematicDeviceDiscoveryService service = (HomematicDeviceDiscoveryService) bundleContext
+                    .getService(discoveryServiceRegistration.getReference());
+            service.deactivate();
+
+            discoveryServiceRegistration.unregister();
+            discoveryServiceRegistration = null;
+            discoveryService = null;
+        }
+    }
+
+    /**
+     * Sets the OFFLINE status for all things of this bridge that has been removed from the gateway.
+     */
+    public void setOfflineStatus() {
```

Comment:
normally, I would expect the ThingHandler to update its state itself and not to have this done from outside. Can't you inform it that its device does not exist anymore?